### PR TITLE
Aura and Slots refactoring

### PR DESCRIPTION
**Pull request by bkchr**
Some further small refactoring of `sc-consensus-slots` and `sc-consensus-aura`:

- `SlotData::slot_duration` now returns a `std::time::Duration` to be more abstract over the implicit "u64 means milliseconds" requirement ;)
- Add `build_aura_worker` function to only create an `AuraWorker` that can be run by the caller.

polkadot companion: https://github.com/paritytech/polkadot/pull/2634

### CODE REVIEW

**Comment by bkchr** [comment]
(This pr is not audited, but also does not require an audit. However, the COMPUTER wants to have a label ;))

File: client/consensus/aura/src/lib.rs [hunk 1]
```diff
@@ -75,7 +75,7 @@ pub use sc_consensus_slots::SlotProportion;
 type AuthorityId<P> = <P as Pair>::Public;
 
 /// Slot duration type for Aura.
-pub type SlotDuration = sc_consensus_slots::SlotDuration<u64>;
+pub type SlotDuration = sc_consensus_slots::SlotDuration<sp_consensus_aura::SlotDuration>;
```
Comment (andresilva) [inline client/consensus/aura/src/lib.rs:78]: Beautiful 😂

File: primitives/timestamp/src/lib.rs [hunk 2]
```diff
@@ -100,8 +106,8 @@ impl From<Timestamp> for u64 {
 	}
 }
 
-impl From<sp_std::time::Duration> for Timestamp {
-	fn from(duration: sp_std::time::Duration) -> Self {
+impl From<Duration> for Timestamp {
```
Comment (andresilva) [inline primitives/timestamp/src/lib.rs:109]: We can maybe remove this one if it's unused.

**Review by andresilva — APPROVED** [review]

File: client/consensus/aura/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply client/consensus/aura/src/lib.rs:78]: Yeah I know :P Some more future refactoring :P

File: client/consensus/aura/src/lib.rs [hunk 3]
```diff
@@ -596,10 +666,10 @@ mod tests {
 					let inherent_data_providers = InherentDataProviders::new();
 					register_aura_inherent_data_provider(
 						&inherent_data_providers,
-						slot_duration.get()
+						slot_duration.slot_duration()
```
Comment (octol) [inline client/consensus/aura/src/lib.rs:669]: Gives the code a little Java flavour ;)

**Review by octol — APPROVED** [review]

**Comment by andresilva** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Pull request merged**